gnomad_qc/v4/assessment/calculate_per_sample_stats.py

```diff
@@ -322,17 +336,35 @@ def create_per_sample_counts_ht(
     :param mt: Input MatrixTable containing variant data. Must have multi-allelic sites
         split.
     :param filter_group_ht: Table containing filter groups for summary stats.
+    :param autosomes_only: Whether to restrict analysis to autosomes only. Default is
+        False.
     :return: Table containing per-sample variant counts.
     """
-    # Add extra Allele Count and Allele Type annotations to variant MatrixTable,
-    # according to Hail standards, to help their computation.
-    variant_ac, variant_types = vmt_sample_qc_variant_annotations(
+    # Add Allele Type annotations to variant MatrixTable.
+    _, variant_types = vmt_sample_qc_variant_annotations(
         global_gt=mt.GT, alleles=mt.alleles
     )
-    mt = mt.annotate_rows(variant_ac=variant_ac, variant_atypes=variant_types)
+    mt = mt.annotate_rows(variant_atypes=variant_types)
+    if autosomes_only:
+        ab_cutoff = 0.9
+        mt = filter_to_autosomes(mt)
+        mt = mt.annotate_entries(
+            GT=hl.if_else(
+                (mt.variant_af > 0.01)
+                & ((mt.AD[1] / mt.DP) > ab_cutoff)
+                & mt.GT.is_het()
+                & ~mt._het_non_ref,
+                hl.call(1, 1),
+                mt.GT,
+            )
+        )
 
     # Annotate the MT with the needed annotations.
     mt = annotate_with_ht(mt, filter_group_ht, filter_missing=True)
```

Comment:
If we're still fixing high ab het, this has to be before the autosome_only argument, because you won't have `af`.